Multiple definition error

I've tried to compile IRAF package from github and I've got multiple definition error.
```/usr/bin/ld: xppcode.o:/home/nastya/iraf/unix/boot/spp/xpp/xppcode.c:109: multiple definition of `errflag'; xppmain.o:/home/nastya/iraf/unix/boot/spp/xpp/xppmain.c:21: first defined here
collect2: error: 
ld returned 1 exit status
make: *** [Makefile:19: sysgen] Error 1```
My operating system is Manjaro Linux. I use gcc (GCC) 10.1.0.

I've also tried to compile this package on other LINUX distros (on Ubuntu, Debian) and everything was fine.

The problem appears only on Arch. I also tried to use different versions of GCC (5.5 and 7.5.0) and it doesn't solve the problem.

I also attach my build.log file.
I appreciate any help and advice.
[build.log](https://github.com/iraf-community/iraf/files/4741080/build.log)

Hello, I applied this patch and I got more multiple definition errors.
```/usr/bin/ld: libf.a(sctabl.o):/home/nastya/iraf/unix/boot/spp/rpp/ratlibf/sctabl.f:5: multiple definition of `cdsmem_'; librpp.a(ratfor.o):/home/nastya/iraf/unix/boot/spp/rpp/rppfor/ratfor.f:83: first defined here
/usr/bin/ld: libf.a(stlu.o):/home/nastya/iraf/unix/boot/spp/rpp/ratlibf/stlu.f:5: multiple definition of `cdsmem_'; librpp.a(ratfor.o):/home/nastya/iraf/unix/boot/spp/rpp/rppfor/ratfor.f:83: first defined here
/usr/bin/ld: libf.a(dsdump.o):/home/nastya/iraf/unix/boot/spp/rpp/ratlibf/dsdump.f:5: multiple definition of `cdsmem_'; librpp.a(ratfor.o):/home/nastya/iraf/unix/boot/spp/rpp/rppfor/ratfor.f:83: first defined here
/usr/bin/ld: libf.a(dsdbiu.o):/home/nastya/iraf/unix/boot/spp/rpp/ratlibf/dsdbiu.f:5: multiple definition of `cdsmem_'; librpp.a(ratfor.o):/home/nastya/iraf/unix/boot/spp/rpp/rppfor/ratfor.f:83: first defined here
collect2: error: ld returned 1 exit status
make: *** [Makefile:19: sysgen] Error 1```
The build.log file is attached.
[build.log](https://github.com/iraf-community/iraf/files/4741623/build.log)

I managed to compile it by adding a flag -fcommon to the CFLAGS in the makefile. But when I started to run tests I've got some errors:
```=================== Failure in numerical-recipes.md:619 with ecl.e ===================

Expected
========
cl> softools
cl> xc -x test_twofft.x numrep.x
cl> task $test_twofft = test_twofft.e
cl> test_twofft
in 1   1.000  2.000  1.000 -1.000  1.500  1.000  0.500  1.000
in 2   0.900  7.500  6.500  5.000  7.500  5.200  5.100  7.700
out 1  7.000 +  0.000i,  1.621 + -0.207i,  1.000 +  3.000i, -2.621 + -1.207i,  1.000 +  0.000i, -2.621 +  1.207i,  1.000 + -3.000i,  1.621 +  0.207i,
out 2 45.400 +  0.000i, -3.064 +  1.117i, -3.200 +  0.000i, -10.14 + -1.683i, -5.400 +  0.000i, -10.14 +  1.683i, -3.200 + 0.000i, -3.064 + -1.117i,

Output
======
cl> softools
cl> xc -x test_twofft.x numrep.x
/usr/bin/ld: numrep.o:/tmp/iraftest.vhnv7rGsL/testroot/numrep.x:5: multiple definition of `mem_'; test_twofft.o:/tmp/iraftest.vhnv7rGsL/testroot/test_twofft.x:11: first defined here
collect2: error: ld returned 1 exit status
cl> task $test_twofft = test_twofft.e
cl> test_twofft
ERROR: Cannot open connected subprocess (./test_twofft.e)
     called as: `cl ()'
Error while reading login.cl file - may need to rebuild with mkiraf
Fatal startup error.  CL dies.

Diff
====
@@ -1,8 +1,10 @@
 cl> softools
 cl> xc -x test_twofft.x numrep.x
+/usr/bin/ld: numrep.o:/tmp/iraftest.vhnv7rGsL/testroot/numrep.x:5: multiple definition of `mem_'; test_twofft.o:/tmp/iraftest.vhnv7rGsL/testroot/test_twofft.x:11: first defined here
+collect2: error: ld returned 1 exit status
 cl> task $test_twofft = test_twofft.e
 cl> test_twofft
-in 1   1.000  2.000  1.000 -1.000  1.500  1.000  0.500  1.000
-in 2   0.900  7.500  6.500  5.000  7.500  5.200  5.100  7.700
-out 1  7.000 +  0.000i,  1.621 + 0.207i,  1.000 +  3.000i, -2.621 + -1.207i,  1.000 +  0.000i, -2.621 +  1.207i,  1.000 + -3.000i,  1.621 +  0.207i,
-out 2 45.400 +  0.000i, -3.064 +  1.117i, -3.200 +  0.000i, -10.140 + -1.683i, -5.400 +  0.000i, -10.140 +  1.683i, -3.200 + 0.000i, -3.064 + -1.117i,
+ERROR: Cannot open connected subprocess (./test_twofft.e)
+     called as: `cl ()'
+Error while reading login.cl file - may need to rebuild with mkiraf
+Fatal startup error.  CL dies.
```

--- a/unix/boot/spp/xpp/xppcode.c
+++ b/unix/boot/spp/xpp/xppcode.c
@@ -106,7 +106,7 @@ int	ntasks = 0;			/* number of tasks in interpreter     */
 int	str_idnum = 0;			/* for generating unique string names */
 int	nbrace = 0;			/* must be zero when "end" is reached */
 int	nswitch = 0;			/* number switch stmts in procedure   */
-int	errflag;
+extern int	errflag;
 int	errhand = NO;			/* set if proc employs error handler  */
 int	errchk = NO;			/* set if proc employs error checking */
 

--- a/unix/hlib/f77.sh
+++ b/unix/hlib/f77.sh
@@ -42,7 +42,7 @@
 
 s=/tmp/stderr_$$
 CC=${CC_f2c:-'gcc'}
-CFLAGS="-I${iraf}include -I${iraf}unix/f2c/libf2c ${XC_CFLAGS} -Wno-maybe-uninitialized -Wno-strict-aliasing -Wno-unknown-warning-option"
+CFLAGS="-I${iraf}include -I${iraf}unix/f2c/libf2c ${XC_CFLAGS} -Wno-maybe-uninitialized -Wno-strict-aliasing -Wno-unknown-warning-option -fcommon"
 EFL=${EFL:-/v/bin/efl}
 EFLFLAGS=${EFLFLAGS:-'system=portable deltastno=10'}
 F2C=${F2C:-${iraf}unix/bin/f2c.e}

